parser/parser_go1.18.go

```diff
@@ -0,0 +1,61 @@
+//go:build go1.18
+// +build go1.18
+
+package parser
+
+import (
+	"go/ast"
+	"reflect"
+
+	"encr.dev/parser/est"
+	schema "encr.dev/proto/encore/parser/schema/v1"
+)
+
+func init() {
+	additionalTypeResolver = go118ResolveType
+}
+
+func go118ResolveType(p *parser, pkg *est.Package, file *est.File, expr ast.Expr) *schema.Type {
+	switch expr := expr.(type) {
+	// Needed for generic types with single generic parameters: `X[Index]` (i.e. `Vector[string]`)
+	case *ast.IndexExpr:
+		return resolveWithTypeArguments(p, pkg, file, expr.X, expr.Index)
+
+	// Needed for generic types with multiple generic parameters: `X[A, B]` (i.e. `Skiplist[string, string]`)
+	case *ast.IndexListExpr:
+		return resolveWithTypeArguments(p, pkg, file, expr.X, expr.Indices...)
+	}
+
+	return nil
+}
+
+func resolveWithTypeArguments(p *parser, pkg *est.Package, file *est.File, ident ast.Expr, typeArguments ...ast.Expr) *schema.Type {
+	// First we need to resolve the parameterized declaration
```

Comment:
Possibly move this comment and the one below up into the doc comment as they're of the "explain what a function does" flavor.